Add example illustrating the use of RAPIDS with MLFlow (#3028)
mlflow/mlflow: examples/rapids/mlflow_project/README.md

> ### Train and Publish Locally With RAPIDS and MLFlow

Reviewer: ```suggestion
### Train and Deploy Models with RAPIDS and MLflow
```

Seems like model training & subsequent deployment with MLflow are the focus of the example? Although correct me if I'm wrong :)

**[RAPIDS](https://rapids.ai/)** is a suite of open source libraries for GPU-accelerated analytics. 

**[RAPIDS cuML](https://github.com/rapidsai/cuml)** matches the scikit-learn API, so it can build on MLFlow's existing support for scikit-learn-like models to support 
persistence and deployment."

> 

Reviewer: Maybe add a sentence describing what our example workflows do, e.g:

```suggestion
The example workflows below train RAPIDs regression models to predict airline flight delays, using MLflow to log models and deploy them as local REST API endpoints for real-time inference.
```

^assuming that such a description is accurate :)

#### Jupyter Notebook Workflow
[Jupyter Notebook](notebooks/rapids_mlflow.ipynb)

#### CLI Based Workflow
1. Create a new conda environment.

>     1. `$ conda create -f envs/conda.yaml`

Reviewer: Shouldn't `mlflow run` create this conda environment for the user? We might be able to remove this step

> 1. Train the model

Reviewer: ```suggestion
1. Train the model
  1. In your shell, run:
   shell script
# Download the file
wget -N https://rapidsai-cloud-ml-sample-data.s3-us-west-2.amazonaws.com/airline_small.parquet
# Launch the job
mlflow run mlflow_project -e hyperopt \
         -P conda-env=$PWD/envs/conda.yaml \
         -P fpath=airline_small.parquet
  1. The above command kicks off RAPIDS model training using MLflow's projects feature. MLflow will create the necessary conda environment for model training and pass parameters specified via `-P` to our training script. See https://mlflow.org/docs/latest/projects.html# for more info
```

Maybe we could switch to something like the above for brevity?

Reviewer: I think it'd also be worth it to try logging results to a local SQLite file (e.g. encourage users to run `MLFLOW_TRACKING_URI=./mlflow-db.sqlite mlflow run ...`). Then we could encourage users to use the model registry when deploying their model (e.g. they can just run `mlflow models serve models:/myRapidsModel/1` to serve their model), as opposed to getting the artifact path via the UI and plugging it into the `mlflow models serve` command - let me know if that makes sense!

### Train and Publish Locally With RAPIDS and MLFlow
**[RAPIDS](https://rapids.ai/)** is a suite of open source libraries for GPU-accelerated analytics. 

> **[RAPIDS cuML](https://github.com/rapidsai/cuml)** matches the scikit-learn API, so it can build on MLFlow's existing support for scikit-learn-like models to support 

Reviewer 2: @drobison00 Can you replace `MLFlow` with `MLflow`?

persistence and deployment."

The example workflows below train RAPIDs regression models to predict airline flight delays, using MLflow to log models and deploy them as local REST API endpoints for real-time inference.

#### Jupyter Notebook Workflow
[Jupyter Notebook](notebooks/rapids_mlflow.ipynb)

#### CLI Based Workflow
1. Download data
    1. `cd examples/rapids/mlflow_project`
        ```shell script
        # Download the file
        wget -N https://rapidsai-cloud-ml-sample-data.s3-us-west-2.amazonaws.com/airline_small.parquet

>         ```

Reviewer: We should include a test for the CLI-based workflow - it should be doable by adding an item to the list of `(directory, args_to_mlflow_run)` tuples in the existing [examples tests](https://github.com/mlflow/mlflow/blob/91dca77a7e850ec25a88dd79736cbab2bfd740a3/tests/examples/test_examples.py#L59). For example, you should be able to add an item like `(os.path.join("rapids", "mlflow_project"), ["-e", "simple"])` and the test will invoke `mlflow run rapids/mlflow_projects -e simple` and verify that it succeeds.

It might be easiest to test this if we update the project so that `fpath` is actually an optional parameter and when unspecified, the training script downloads the data via logic like:

```
import requests
import tempfile

# Data will ultimately be written to the file whose path is stored in fname
_, fname = tempfile.mkstemp()
url = "https://rapidsai-cloud-ml-sample-data.s3-us-west-2.amazonaws.com/airline_small.parquet"
airline_data_response = requests.get(url)
with open(fname , 'wb') as handle:
    handle.write(airline_data_response.content)
```

We can defer this to a follow-up PR, but would be good to have - otherwise, this example may break in the future.

Reviewer: Ah, just discussed offline with @harupy - looks like our underlying CI provider (Github actions) doesn't support GPUs, so we unfortunately can't actually test this example - feel free to disregard the above comment.

1. Set MLflow tracking uri
    1. ```shell script
        export MLFLOW_TRACKING_URI=sqlite:////tmp/mlflow-db.sqlite
       ```
1. Train the model using a single run.
    1. ```shell script
       # Launch the job

>        mlflow run . -b local -e hyperopt \

Reviewer: We should be able to remove the `-b local` from this `mlflow run` invocation and the one below, as the default backend for project execution is "local"

                --experiment-name RAPIDS-CLI \
                -P max_depth=10 -P max_features=0.75 -P n_estimators=500 \

>                 -P conda-env=$PWD/envs/conda.yaml \

Reviewer: We no longer need to pass conda-env here right?
```suggestion
```

Also, I like the idea of having an easy-to-run "simple" entry point - could we add default values for the max_depth, max_features, n_estimators params in the MLproject file, as shown in [the docs](https://mlflow.org/docs/latest/projects.html#mlproject-file), and mention that the user can simply run `mlflow run . -e simple`?
